python/python/lancedb/embeddings/bedrock.py

```diff
@@ -87,21 +89,29 @@ def ndims(self):
         # TODO: fix hardcoding
         if self.name == "amazon.titan-embed-text-v1":
             return 1536
-        elif self.name in {"cohere.embed-english-v3", "cohere.embed-multilingual-v3"}:
+        elif self.name in [
+            "amazon.titan-embed-text-v2:0",
+            "cohere.embed-english-v3",
+            "cohere.embed-multilingual-v3",
+        ]:
+            # TODO: "amazon.titan-embed-text-v2:0" model supports dynamic ndims
             return 1024
         else:
-            raise ValueError(f"Unknown model name: {self.name}")
+            raise ValueError(f"Model {self.name} not supported")
 
     def compute_query_embeddings(
         self, query: str, *args, **kwargs
     ) -> List[List[float]]:
-        return self.compute_source_embeddings(query)
+        return self.compute_source_embeddings(query, input_type=self.query_input_type)
```

Comment:
Non-blocking: Should this use the same `kwargs.get("input_type") or self.query_input_type` logic as in `compute_source_embeddings`? 
This logic is perfectly fine for the typical search/retrieval case. I'm much less familiar with the other use cases for vector databases but I would imagine other input types (such as `"clustering"`) might be useful.